internal/providervalidation/providervalidation.go

```diff
@@ -183,6 +186,15 @@ func ValidateProvider(ctx context.Context, st blob.Storage, opt Options) error {
 		return errors.Wrap(err, "error validating concurrency")
 	}
 
+	log(ctx).Infof("Validating blob recreation...")
+
+	if !opt.SupportIdempotentCreates {
+		err := st.PutBlob(ctx, "dummy_id", gather.FromSlice([]byte{99}), blob.PutOptions{DoNotRecreate: true})
+		if err == nil {
```

Comment:
can we introduce and expect `ErrUnsupportedPutBlobOption` here?

Reply:
Yes, thanks for the suggestion @jkowalski. Added this new type new and old opts.